### ISSUE TITLE
Does DG support Call graph with Function Pointer?

### ISSUE THREAD

**Issue by Marvinmw** [mchalupa/dg#306]
Does DG support Call graph with Function Pointer?

**Comment by mchalupa** [comment]
Yes, it does. The relevant object is `callGraph` in the class PointerGraph.

**Comment by Marvinmw** [comment]
> Yes, it does. The relevant object is `callGraph` in the class PointerGraph.

Thanks for your reply. I try using DG. But I am curious that what is LLVMNode? How do I mapy BasicBlock, Function in llvm libraryt to DG data type?

**Comment by mchalupa** [comment]
It depends on what information about LLVM you need. If you need dependencies between instructions, you can use LLVMNode from the LLVMDependenceGraph. If you need information about pointers, you need to query pointer analysis, etc. So what information are you interested in?

**Comment by Marvinmw** [comment]
> It depends on what information about LLVM you need. If you need dependencies between instructions, you can use LLVMNode from the LLVMDependenceGraph. If you need information about pointers, you need to query pointer analysis, etc. So what information are you interested in?

Thanks a lot. Now I want to use dg to develop in my project. Is there some way to configure it as the third library in my project?

**Comment by mchalupa** [comment]
Sure, dg does not generate a cmake config file yet, but you can configure it manually, e.g., like here: https://github.com/staticafi/sbt-instrumentation/blob/master/CMakeLists.txt#L50 (or if you use autotools, you just need to set the right paths there).

**Comment by mchalupa** [comment]
another possible source of information: https://github.com/staticafi/sbt-slicer/blob/master/CMakeLists.txt#L64

**Comment by Marvinmw** [comment]
> another possible source of information: https://github.com/staticafi/sbt-slicer/blob/master/CMakeLists.txt#L64

thanks. I am trying.
